### PR TITLE
Accept resource names for hostdevicenetwork either with or without prefix

### PR DESCRIPTION
**Pull request by almaslennikov**
Currently network operator doesn't allow deploying host device networks if the resource name start with "nvidia.com" prefix. The solution is to normalize all resource names for hostdevicenetwork state to start with a common prefix.

### CODE REVIEW

File: pkg/state/state_hostdevice_network_test.go [hunk 1]
```diff
@@ -29,6 +31,14 @@ import (
 	"github.com/Mellanox/network-operator/pkg/utils"
 )
 
+func checkResourceNameAnnotation(obj *unstructured.Unstructured) {
+	annotations := obj.Object["metadata"].(map[string]interface{})["annotations"].(map[string]interface{})
+	resourceName := annotations["k8s.v1.cni.cncf.io/resourceName"].(string)
+
+	Expect(strings.HasPrefix(resourceName, resourceNamePrefix)).To(Equal(true))
```
Comment (ykulazhenkov) [inline pkg/state/state_hostdevice_network_test.go:38]: nit: Equal(true) -> BeTrue()

File: pkg/state/state_hostdevice_network_test.go [hunk 2]
```diff
@@ -29,6 +31,14 @@ import (
...
+	Expect(strings.Count(resourceName, resourceNamePrefix)).To(Equal(1))
```
Comment (ykulazhenkov) [inline pkg/state/state_hostdevice_network_test.go:39]: nit: Expect(strings.Count(resourceName, resourceNamePrefix)).To(Equal(1)) > Expect(resourceName).To(HavePrefix(resourceNamePrefix)) ?

**Review by moshe010 — APPROVED** [review]

**Pull request merged**